c_binding/lsm_datatypes.hpp

```diff
@@ -276,6 +276,7 @@ struct LSM_DLL_LOCAL _lsm_disk {
     const char *location;
     int32_t rpm;
     lsm_disk_link_type link_type;
+    char *plugin_data; /**< Reserved for the plugin to use, added in 1.9 */
```

Comment:
Growing the size of a presumably public `struct`. I think this should be safe in case new struct members are added at the end. No objections with the added functions.

Reply:
Thanks for taking a look.  The structure is also opaque, so code outside the library has no knowledge of internals.